lib/c-api/build.rs

```diff
@@ -419,14 +419,20 @@ fn build_inline_c_env_vars() {
         L = shared_object_dir.clone(),
     );
 
-    // Add `-lwasmer_c_api` and, on Linux,
-    // `-rpath=<shared_object_dir>`, to the linker argument.
     println!(
-        "cargo:rustc-env=INLINE_C_RS_LDFLAGS=-lwasmer_c_api,{rpath}",
-        rpath = if cfg!(target_os = "linux") {
-            format!("-rpath={}", shared_object_dir)
+        "cargo:rustc-env=INLINE_C_RS_LDFLAGS={shared_object_dir}/{lib}",
+        shared_object_dir = shared_object_dir,
+        lib = if cfg!(target_os = "windows") {
+            "wasmer_c_api.dll".to_string()
         } else {
-            "".to_string()
-        },
+            format!(
+                "libwasmer_c_api.{ext}",
+                ext = if cfg!(target_os = "macos") {
+                    "dylib".to_string()
+                } else {
+                    "so".to_string()
+                }
+            )
+        }
```

Comment:
This can be much simpler this way:

```suggestion
        lib = if cfg!(target_os = "windows") {
            "wasmer_c_api.dll".to_string()
        } else if cfg!(target_os = "macos") {
            "libwasmer_c_api.dylib".to_string()
        }
        else {
            "libwasmer_c_api.so".to_string()
        }
```